### PR TITLE
VEN-1394 | Fix end date calculation for terminated leases

### PR DESCRIPTION
**Pull request by a-rmz**
### Closes :no_good_woman:
**[VEN-1394](https://helsinkisolutionoffice.atlassian.net/browse/VEN-1394):** In admin UI it is not possible to terminate lease while season is over

### Automated tests :gear:️
```shell
$ pytest leases/tests/test_lease_mutations.py
```

### CODE REVIEW

File: leases/models.py [hunk 1]
```diff
@@ -61,6 +61,18 @@ class Meta:
     def order(self):
         return self.orders.filter(order_type=OrderType.LEASE_ORDER).first()
 
+    @property
+    def has_started(self):
+        return today().date() >= self.start_date
+
+    @property
+    def has_ended(self):
+        return today().date() > self.end_date
```
Comment (tuomas777) [inline leases/models.py:70]: Do these also the current timezone into account?

File: leases/utils.py [hunk 2]
```diff
@@ -159,6 +159,34 @@ def calculate_winter_storage_lease_end_date() -> date:
...
+
+    # If the lease is terminated after its season has ended, the end date will be set
+    # to the date when the season ends.
+    elif lease.has_ended:
+        if isinstance(lease, BerthLease):
+            return calculate_season_end_date(lease.start_date)
+        else:  # WinterStorageLease
+            return calculate_winter_season_end_date(lease.start_date)
+
+
```
Comment (tuomas777) [inline leases/utils.py:189]: What if none of the `if`s matches, should this then return `None` as it seems to do now? Then at least the return type hint is slightly wrong.

Comment (a-rmz) [reply leases/utils.py:189]: Good point, this last case returns the default end date for the lease, so it should return it in the last case.

File: leases/models.py (hunk 1 above)
Comment (a-rmz) [reply leases/models.py:70]: Hmmm 🤔 well this calculations are done only by and for the system, so it should use the timezone set by the app? I'm not 100% sure but I guess yes.

Comment (tuomas777) [reply leases/models.py:70]: For some reason I wasn't sure about that dateutil's today(), but it seems to work correctly, so you are most probably absolutely correct 🙂

**Review by tuomas777 — APPROVED** [review]
LGTM 👍 ⛵

File: leases/models.py (hunk 1 above)
Comment (a-rmz) [reply leases/models.py:70]: It has worked so far but I never really thought about it 🙈

**Pull request merged**